sklearn/decomposition/_sparse_pca.py

```diff
@@ -388,9 +484,20 @@ def fit(self, X, y=None):
             return_n_iter=True)
         self.components_ = Vt.T
 
-        components_norm = np.linalg.norm(
-            self.components_, axis=1)[:, np.newaxis]
-        components_norm[components_norm == 0] = 1
-        self.components_ /= components_norm
+        # requires the mean to be subtracted
+        if not self.normalize_components:
+            # should be removed in version 0.22
```

Comment:
I think this is using an outdated version of SparsePCA. The only change I would expect in fit in this diff is to additional compute `explained_variance_` not changes related to `normalize_components`.